content/courses/mobile/solana-mobile-dapps-with-expo.md

```diff
@@ -923,61 +1130,65 @@ export interface NFTContextState {
 }
 ```
 
-The state flow here is: `connect`, `fetchNFTs`, and then `createNFT`. We'll walk
+The state flow here is: connect, fetchNFTs, and then createNFT. We'll walk
 through the code for each of them and then show you the entire file at the end:
 
 1. `connect` - This function will connect and authorize the app, and then store
    the resulting `publicKey` into the state.
 
-   ```tsx
-   const connect = () => {
-     if (isLoading) return;
+```tsx
+const connect = async () => {
+  try {
+    if (isLoading) return;
 
-     setIsLoading(true);
-     transact(async wallet => {
-       const auth = await authorizeSession(wallet);
-       setAccount(auth);
-     }).finally(() => {
-       setIsLoading(false);
-     });
-   };
-   ```
+    setIsLoading(true);
+    await transact(async wallet => {
+      const auth = await authorizeSession(wallet);
+      setAccount(auth);
+    });
+  } catch (error) {
+    console.log(error);
+  } finally {
+    setIsLoading(false);
+  }
+};
+```
 
 2. `fetchNFTs` - This function will fetch the NFTs using Metaplex:
 
-   ```tsx
-   const fetchNFTs = async () => {
-     if (!metaplex || !account || isLoading) return;
-
-     setIsLoading(true);
-
-     try {
-       const nfts = await metaplex.nfts().findAllByCreator({
-         creator: account.publicKey,
-       });
-
-       const loadedNFTs = await Promise.all(
-         nfts.map(nft => {
-           return metaplex.nfts().load({ metadata: nft as Metadata });
-         }),
-       );
-       setLoadedNFTs(loadedNFTs);
-
-       // Check if we already took a snapshot today
-       const nftOfTheDayIndex = loadedNFTs.findIndex(nft => {
-         return formatDate(new Date(Date.now())) === nft.name;
-       });
-
-       if (nftOfTheDayIndex !== -1) {
-         setNftOfTheDay(loadedNFTs[nftOfTheDayIndex]);
-       }
-     } catch (error) {
-       console.log(error);
-     } finally {
-       setIsLoading(false);
-     }
-   };
-   ```
+```tsx
+const fetchNFTs = async () => {
+  if (!metaplex || !account || isLoading) return;
+
+  setIsLoading(true);
+
+  try {
+    const nfts = await metaplex.nfts().findAllByCreator({
+      creator: account.publicKey,
+    });
+
+    const loadedNFTs = await Promise.all(
+      nfts.map(nft => {
+        return metaplex.nfts().load({ metadata: nft as Metadata });
+      }),
+    );
+    setLoadedNFTs(loadedNFTs);
+
+    // Check if we already took a snapshot today
+    const nftOfTheDayIndex = loadedNFTs.findIndex(nft => {
+      return formatDate(new Date(Date.now())) === nft.name;
+    });
+
+    if (nftOfTheDayIndex !== -1) {
```

Comment:
'Does not equal -1' is very old-style JS. Why not do `const nftOfTheDay = loadedNFTs.find((loadedNft) => { ... })` and then `if ( ! loadedNft ) {}`?

Reply:
Sorry, the double negative confused me (which is oddly why I'm making this). `if ( loadedNft )` then.

Reply:
Ohh, got it! Doing the changes

Reply:
Okay, that was resolved in the last commit I made